### PR TITLE
feat: allow overriding stream implementation

### PR DESCRIPTION
**Pull request by ronag**
Fixes: https://github.com/mcollina/undici/issues/17

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -100,6 +99,11 @@ class Client extends EventEmitter {
 
     this.url = url
 
+    this[kStream] = opts.stream || {
+      finished: require('end-of-stream'),
```
Comment (mcollina) [inline lib/client.js:103]: I would get this from readable-stream anyway, it's there.

**Comment by ronag** [comment]
PTAL

**Comment by mcollina** [comment]
CI is failing :(

**Comment by ronag** [comment]
Travis was building from old version, guess it still has some kind of race problems. Forced a restart.

**Comment by ronag** [comment]
I'm confused... travis is failing on a line of code that does not exist?

**Comment by ronag** [comment]
Ah, travis merges against master.

**Comment by mcollina** [comment]
there are still conflicts

**Comment by ronag** [comment]
new conflicts resolved

**Comment by mcollina** [comment]
lol, more conflicts

**Comment by ronag** [comment]
fixed again

**Comment by mcollina** [comment]
still conflicts :(

**Comment by ronag** [comment]
argh, fixed again

**Pull request merged**